sdks/java/core/src/test/java/org/apache/beam/sdk/options/PipelineOptionsFactoryTest.java

```diff
@@ -1797,6 +1799,44 @@ public void serialize(JacksonIncompatible jacksonIncompatible, JsonGenerator jso
     }
   }
 
+  @Test
+  public void testPipelineOptionsFactoryFromProperties() {
+    assertEquals("testAppName", PipelineOptionsFactory.fromProperties(new Properties() {{
```

Comment:
Add a string key with non string value to the properties and also a non string key and string value and validate that they aren't used.